.github/workflows/preview-pr-build.yaml

```
@@ -5,53 +5,76 @@ on:
     types: [opened, synchronize, reopened, ready_for_review]
     paths:
       - 'content/**'
-      - 'config/**'
       - 'static/**'
-      - 'themes/**'
-      - 'package.json'
 
 jobs:
   deploy:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-latest    
     steps:
-      - name: 检出主仓库代码
+      - name: 克隆 Next-MtF-wiki 仓库
         uses: actions/checkout@v4
         with:
-          ref: ${{ github.event.pull_request.base.ref }}
+          repository: project-trans/Next-MtF-wiki
           fetch-depth: 0
           submodules: true
 
-      - uses: DeterminateSystems/nix-installer-action@main
-
-      - name: 检出 PR 修改的部分
+      - name: 更新 source submodule 为 PR 的 ref 版本
         run: |
+          cd source
+          git remote set-url origin https://github.com/${{ github.event.pull_request.head.repo.full_name }}.git
           git fetch origin +refs/pull/${{ github.event.pull_request.number }}/merge
           git checkout -qf FETCH_HEAD
-          git submodule update
-
-      # - name: 配置 baseURL
-      #   run: |
-      #     sed -i 's|baseURL = "https://mtf.wiki/"|baseURL = "/"|' config/_default/config.toml
+          cd ..
+          git add source
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git commit -m "Update source submodule to PR #${{ github.event.pull_request.number }}" || true
 
-      - name: Install Node.js
+      - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: '22'
-          cache: 'npm'
-
-      - name: Install npm dependencies
-        run: npm install
-
-      - name: Build site
-        run: nix run github:nixos/nixpkgs/nixos-24.11#hugo -- --minify --enableGitInfo --baseURL "/"
-
+          node-version: '20'
+          
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: '10.10.0'
+          
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+          
+      - name: Setup pnpm cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
```

```suggestion
          node-version: '22'
          cache: 'pnpm'
          
      - name: Setup pnpm
        uses: pnpm/action-setup@v4
        with:
          version: '10.10.0'
```

Node.js 使用当前的 LTS：22，cache 'pnpm' 即可自动配置缓存，无需手动配置